data/block/miniBlockHeader.go

```diff
@@ -164,6 +164,56 @@ func (m *MiniBlockHeader) setMiniBlockHeaderReserved(mbhr *MiniBlockHeaderReserv
 	return nil
 }
 
+// GetIndexOfFirstTxProcessed returns index of the first transaction processed in the miniBlock
+func (m *MiniBlockHeader) GetIndexOfFirstTxProcessed() int32 {
+	miniBlockHeaderReserved, err := m.getMiniBlockHeaderReserved()
+	if err != nil || miniBlockHeaderReserved == nil {
+		return 0
+	}
+
+	return miniBlockHeaderReserved.IndexOfFirstTxProcessed
+}
+
+// GetIndexOfLastTxProcessed returns index of the last transaction processed in the miniBlock
+func (m *MiniBlockHeader) GetIndexOfLastTxProcessed() int32 {
+	miniBlockHeaderReserved, err := m.getMiniBlockHeaderReserved()
+	if err != nil || miniBlockHeaderReserved == nil {
+		return int32(m.TxCount) - 1
```

Comment:
if 0 tx count you would get -1. expected? do you treat this when used?

Reply:
There could not exists empty mini blocks added in block body. At least 1 tx should be in each mini block added by proposer in block body and signed by consensus group as well.